framework/wazuh/core/agent.py

```diff
@@ -1756,3 +1176,83 @@ def get_rbac_filters(system_resources=None, permitted_resources=None, filters=No
         negate = True
 
     return {'filters': filters, 'rbac_negate': negate}
+
+
+def agents_padding(result, agent_list):
+    """This function remove agent 000 from agent_list and
+    it transforms the format of the agent ids to the general format
+
+    Parameters
+    ----------
+    result : AffectedItemsWazuhResult
+    agent_list : list
+        List of agent's IDs
+
+    Returns
+    -------
+    Formatted agent list
+    """
+    agent_list = [str(agent).zfill(3) for agent in agent_list]
+    if '000' in agent_list:
+        result.add_failed_item(id_='000', error=WazuhError(code=1703))
+        agent_list.remove('000')
+
+    return agent_list
+
+
+def core_upgrade_agents(agents_chunk, command='upgrade_result', wpk_repo=None, version=None,
```

Comment:
Why are we using the same core function to send messages to upgrade module and task module? This function should only be in charge of communicating with upgrade module, for either upgrade_result or start upgrades.

Reply:
In order to get the upgrade result, the task socket is needed.